fakeredis/commands_mixins/json_mixin.py

```diff
@@ -0,0 +1,605 @@
+"""Command mixin for emulating `redis-py`'s JSON functionality."""
+
+# Future Imports
+from __future__ import annotations
+
+# Standard Library Imports
+import json
+import re
+from functools import partial
+from itertools import (
+    chain,
+    repeat,
+    starmap,
+)
+from operator import (
+    attrgetter,
+    methodcaller,
+)
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    ForwardRef,
+    Optional,
+    Union,
+)
+
+# Third-Party Imports
+from redis.commands.json.commands import JsonType
+from typing_extensions import Literal
+
+# Package-Level Imports
+from fakeredis import _msgs as msgs
+from fakeredis._commands import (
+    CommandItem,
+    Key,
+    command,
+)
+from fakeredis import _helpers as helpers
+
+if TYPE_CHECKING:
+    # Package-Level Imports
+    from fakeredis._fakesocket import FakeSocket
+else:
+    FakeSocket = ForwardRef("JSONCommandsMixin")
+
+try:
+    from jsonpath_ng import jsonpath
+    from jsonpath_ng.ext import parse as parse_jsonpath
+except ImportError:
+
+    jsonpath = None
+
+    def parse_jsonpath(*_: Any, **__: Any) -> Any:
+        """Raise an error."""
+        raise helpers.SimpleError("Optional JSON support not enabled!")
+
+
+path_pattern: re.Pattern = re.compile(r"^((?<!\$)\.|(\$\.$))")
+
+
+def format_jsonpath(path: Union[str, bytes]) -> str:
+    """Format the supplied JSON path value."""
+    if isinstance(path, bytes):
+        path = path.decode()
+
+    return path_pattern.sub("$", path)
+
+
+def null_op(obj: JsonType, *_: Any, **__: Any) -> None:
+    """Return the supplied object, completely unaltered."""
+    return obj
+
+
+class JSONObject:
+    """Argument converter for JSON objects."""
+
+    DECODE_ERROR = msgs.WRONGTYPE_MSG
+    ENCODE_ERROR = msgs.WRONGTYPE_MSG
+
+    @classmethod
+    def valid(cls, value: Any) -> bool:
+        """Determine if the supplied value is a "valid" `JSONObject`."""
+        raise NotImplementedError
+
+    @classmethod
+    def decode(cls, value: bytes) -> Any:
+        """Deserialize the supplied bytes into a valid Python object."""
+        return json.loads(value)
+
+    @classmethod
+    def encode(cls, value: Any) -> bytes:
+        """Serialize the supplied Python object into a valid, JSON-formatted
+        byte-encoded string."""
+        raise NotImplementedError
+
+
+class JSONCommandsMixin:
+    """`CommandsMixin` for enabling RedisJSON compatibility in `fakeredis`."""
+
+    def __init__(
+        self: FakeSocket,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        pass
```

Comment:
Given the lack of unity on returning `str` or `bytes` between any given JSON library this could get hairy _fast_, not to mention the fact that we'd then be _explicitly_ adding an inherited `_encode` method to the `FakeSocket` class.

Having taken your suggestion to revert to the standard library's `json` module instead of `orjson`, it doesn't seem worth the potential future headache. I've added the requested `super` call.